ding_user.module

```diff
@@ -478,12 +480,73 @@ function ding_user_default_authname($name) {
   // it.
   require_once DRUPAL_ROOT . '/' . variable_get('password_inc', 'includes/password.inc');
 
-  /* A sha512 is 64 bytes, which becomes 128 hexadecimal chars, which is
-  exactly the size of the authname field in the database. So we ask hash
-  for binary data and lets _password_base64_encode base64 encode it to 86
-  characters, leaving enough room for our suffix. */
+  // A sha512 is 64 bytes, which becomes 128 hexadecimal chars, which is
+  // exactly the size of the authname field in the database. So we ask hash
+  // for binary data and lets _password_base64_encode base64 encode it to 86
+  // characters, leaving enough room for our suffix.
   $hash = hash('sha512', drupal_get_private_key() . $name, TRUE);
-  return _password_base64_encode($hash, drupal_strlen($hash)) . '@ding_user';
+
+  // Hence the strlen().
+  $authname = _password_base64_encode($hash, strlen($hash)) . '@ding_user';
+
+  _ding_user_update_authname($authname);
+
+  return $authname;
+}
+
+/**
+ * Updates authname for existing users.
+ *
+ * Brings compatibility of user accounts across various php versions.
+ * Issue caused by drupal_strlen() -> mb_strlen(), whose
+ * behavior was changed somewhat in php 5.4+
+ *
+ * @see http://php.net/releases/NEWS_5_4_0_beta2.txt
+ * - "Ill-formed UTF-8 check for security enhancements."
+ */
+function _ding_user_update_authname($authname) {
```

Comment:
Uhh these changes make my hands shake.. We have had enormous issues with this previously :|